app/views/assignments/_files_form.html.erb

```diff
@@ -0,0 +1,155 @@
+        a team.</p>
+      <%= f.check_box :team_subs, data: {toggle: "toggle", on: "Yes", off: "No"} %>
+    </div>
+    <div class="form-group col-sm-6">
+      <%= f.label :points_available %>
+      <p>
+        <span class="files-type questions-type">This is the percentage of the final grade that this
+          assignment is worth.  Default: same as previous assignment.</span>
+        <span class="exam-type">This is the percentage of the final grade
+          that this exam is worth.</span></p>
+      <%= f.text_field :points_available, size: 10, class: 'form-control numeric' %>
+    </div>
+  </div>
+
+  <div class="row">
+    <div class="form-group col-sm-6">
+      <%= f.label :due_date do %>
+      <span class="files-type questions-type">Due date</span>
+      <% end %>
+      <p>Assignment should not be submitted after this time. Default: one week from today.</p>
+      <div>
+        <%= f.text_field :due_date, class: 'form-control datetime-picker' %>
+      </div>
+    </div>
+    <div class="form-group col-sm-6 files-type questions-type">
+      <%= f.label :available_date %>
+      <p>Assignment will not be visible until this time. Default: today.</p>
+      <div>
+        <%= f.text_field :available, class: 'form-control datetime-picker' %>
+      </div>
+    </div>
+  </div>
+
+  <div class="for-type row">
+    <div class="form-group col-sm-6 files-type questions-type">
+      <%= f.label :request_time_taken, "Request time taken?" %>
+      <p>If selected, ask students on each submission how long they've
+        worked on the assignment.</p>
+      <%= f.check_box :request_time_taken, checked: asgn.request_time_taken,
+          data: {toggle: "toggle", on: "Yes", off: "No"} %>
+    </div>
+  </div>
+  <div class="form-group for-type">
+    <div class="files-type questions-type">
+      <p><strong>Late Penalty</strong></p>
+      <%= render 'lateness/picker', within_course: true, f: f %>
+    </div>
+  </div>
+  <div class="form-group">
+    <div>
+      <p><strong>Graders</strong></p>
+      <div id="files-graders" class="graders-list">
+        <%= f.fields_for :graders, asgn.graders_ordered do |gr| %>
+          <div class="panel panel-default nested-fields">
+            <div class="panel-heading">
+              <% if gr.object.grades.empty? %>
+                <div class="pull-right">
+                  <%= link_to_remove_association "Remove", gr, class: 'btn btn-danger' %>
+                </div>
+              <% end %>
+              <p><strong><%= gr.object.type %></strong></p>
+              <div class="clearfix"></div>
+            </div>
+            <div class="panel-body">
+              <%= render "graders/#{gr.object.type.underscore}.html.erb", f: gr %>
+            </div>
+          </div>
+        <% end %>
+      </div>
+
+      <%= link_to_add_association("Add Grader", f, :graders, partial: 'graders/form',
+                                  "data-association-insertion-node" => '#files-graders',
```

Comment:
Rails question: there are four kinds of parameters being passed to this function call -- simple values like strings and identifiers, symbols like `:graders`, symbol/value mappings like `partial: '...'`, and...what are these string => string things?  Also, what is `link_to_add_association`?